numpy/core/memmap.py

```diff
@@ -242,8 +242,10 @@ def __new__(subtype, filename, dtype=uint8, mode='r+', offset=0,
                 size = bytes // _dbytes
                 shape = (size,)
             else:
-                if not isinstance(shape, tuple):
+                if not hasattr(shape, '__iter__'):
```

Comment:
I would suggest using the same pattern as `normalize_axis_tuple` (I think we may have another pattern around).  You are adding `int` support, so should have a `.. versionchanged::` tag (and release note if you like).

(That will allow `np.array(5)` as a shape.  Not that that should matter in practice, just think it may make sense to reuse a pattern we use in a few places.)

The tuple check is fine, but I suspect not worth it if it is already a tuple calling `tuple` is a no-op (except for subclasses maybe).